ckanext/geodatagov/tests/test_waf-collection.py

```diff
@@ -116,20 +116,30 @@ def run_import(self):
     def test_sample1(self):
```

Comment:
This test seems to be making a lot of assertions, essentially testing multiple things at once. When this test fails, it's difficult to see what specific feature is actually broken. Ideally, it would be really clear from the test name and failure description what is failing and why.

Can you add a docstring that explains what we're testing? For example:

> Integration test: given a WAF-collection harvest source with a single dataset, when the source is harvested:
> - the import phase returns a the child dataset
> - the parent dataset is created with `collection_metadata=true` to signal it is a parent dataset
> - the parent dataset is created with the correct title
> - the child dataset is created with the `collection_package_id` pointing to the parent Id
> - the child dataset is created with the correct title

Just to drive the point home, :point_up: notice how each of these could be its own test.

Reply:
... and notice how this mirrors our BDD Acceptance Criteria (GIVEN ... WHEN ... THEN ...)

Reply:
I [updated the PR](https://github.com/GSA/ckanext-geodatagov/pull/132/commits/23299dbe9dba3a55e2c00d6008a9a1a20577274f) with you suggestions. I think it's clearer now